pkg/kubectl/cmd/util/cmdtools.go

```diff
@@ -85,7 +85,7 @@ func defaultEnvShell() []string {
 	return []string{shell, flag}
 }
 
-func defaultEnv(cmdName CmdName, envs []string) ([]string, bool) {
+func defaultEnv(cmdName CmdName, envs []string, diffViewer string) ([]string, bool) {
```

Comment:
I don't like the style...

You may need to refactor related code.
I think you can try something like:
```
func Complete() {
  // env = GetEnv()
  // fvalue = GetFlagValue()
  // validate env and fvalue, and determine what diff to use
  // o.DiffTool = ...
}

func RunApplyDiffApplied() {
  // launch the cmd tool using o.DiffTool
}
```

If it doesn't make sense to you, let me know.